runtime/lua/vim/_meta/misc.lua

```diff
@@ -14,3 +14,23 @@
 --- @param ... any
 --- @return any
 function vim.call(func, ...) end
+
+--- @class vim.context.mods
+--- @field sandbox? boolean
+--- @field noautocmd? boolean
+--- @field hide? boolean
+--- @field horizontal? boolean
+--- @field keepalt? boolean
+--- @field keepjumps? boolean
+--- @field keepmarks? boolean
+--- @field keeppatterns? boolean
+--- @field lockmarks? boolean
+---
+--- @field buf? integer
+--- @field win? integer
+
+--- @generic R1, R2, R3, R4
+--- @param mods vim.context.mods
+--- @param f fun(): R1, R2, R3, R4
+--- @return R1, R2, R3, R4
+function vim.__with(mods, f) end
```

Comment:
```suggestion
function vim._with(mods, f) end
```
?

Reply:
No, this is the function signature for the `vim.__with` function that is called in `vim._with` function. In hindsight maybe the naming isn't great lol

Reply:
Nothing that can't be addressed with a comment.

Reply:
I just changed the name to `_with_c` to avoid writing a comment instead